docs/code-examples/roundtrips.py

```diff
@@ -0,0 +1,249 @@
+
+    args = parser.parse_args()
+
+    build_env = os.environ.copy()
+    if "RUST_LOG" in build_env:
+        del build_env["RUST_LOG"]  # The user likely only meant it for the actual tests; not the setup
+
+    if args.no_py_build:
+        print("Skipping building python rerun-sdk - assuming it is already built and up-to-date!")
+    else:
+        print("----------------------------------------------------------")
+        print("Building rerun-sdk for Python…")
+        start_time = time.time()
+        returncode = subprocess.Popen(["just", "py-build", "--quiet"], env=build_env).wait()
+        assert returncode == 0, f"Python rerun-sdk build failed with exit code {returncode}"
+        elapsed = time.time() - start_time
+        print(f"rerun-sdk for Python built in {elapsed:.1f} seconds")
+        print("")
+
+    if args.no_cpp_build:
+        print("Skipping cmake configure & build for rerun_c & rerun_cpp - assuming it is already built and up-to-date!")
+    else:
+        print("----------------------------------------------------------")
+        print("Build rerun_c & rerun_cpp…")
+        start_time = time.time()
+        os.makedirs("build", exist_ok=True)
+        build_type = "Debug"
+        if args.release:
+            build_type = "Release"
+        configure_args = ["cmake", f"-DCMAKE_BUILD_TYPE={build_type}", "-DCMAKE_COMPILE_WARNING_AS_ERROR=ON", ".."]
+        print("> ${subprocess.list2cmdline(configure_args)}")
+        returncode = subprocess.Popen(
+            configure_args,
+            env=build_env,
+            cwd="build",
+        ).wait()
+        assert returncode == 0, f"configuring cmake failed with exit code {returncode}"
+        cmake_build("rerun_sdk", args.release)
+        elapsed = time.time() - start_time
+        print(f"rerun-sdk for C++ built in {elapsed:.1f} seconds")
+        print("")
+
+    if len(args.example) > 0:
+        examples = args.example
+    else:
+        dir = os.path.dirname(__file__)
+        files = [f for f in listdir(dir) if isfile(join(dir, f))]
+        examples = [
+            filename
+            for filename, extension in [os.path.splitext(file) for file in files]
+            if extension in (".cpp", ".py", ".rs")
+        ]
+
+    examples = list(set(examples))
+    examples.sort()
+
+    for example in examples:
+        example_opt_out = opt_out.get(example, [])
+
+        if "rust" not in example_opt_out:
+            rust_output_path = run_roundtrip_rust(example, args.release, args.target, args.target_dir)
+
+            if "py" not in example_opt_out:
+                python_output_path = run_roundtrip_python(example)
+                run_comparison(python_output_path, rust_output_path, args.full_dump)
+
+            if "cpp" not in example_opt_out:
+                cpp_output_path = run_roundtrip_cpp(example, args.release)
+                run_comparison(rust_output_path, cpp_output_path, args.full_dump)
```

Comment:
Even if we opt out of *comparing* I suspect it's still helpful to do something like run the example and assert the RRD file is generated and not empty?

Reply:
Right! Had the same thought earlier, let's go for it.